File: Examples/Word/VB/NetOffice COMAddin Examples/06 Custom Addin Object/TimeComponent.vb
Imports System.Runtime.InteropServices
Imports Word06AddinVB4

<ComVisible(True), InterfaceType(ComInterfaceType.InterfaceIsDual), Guid("CD610128-44B7-4AFD-AF34-F8AD9FF1BD8E")>
Public Interface ITimeComponent

    <DispId(1)>
    Function GetTime() As String

End Interface

Public Class TimeComponent
    Implements ITimeComponent

    Public Function GetTime() As String Implements ITimeComponent.GetTime
        Return DateTime.Now.ToString()
    End Function

End Class

File: Examples/Word/C#/NetOffice COMAddin Examples/06 Custom Addin Object/TimeComponent.cs
using System;
using System.Runtime.InteropServices;

namespace Word06AddinCS4
{
    [ComVisible(true), InterfaceType(ComInterfaceType.InterfaceIsDual), Guid("ED27733D-DA7E-4674-804A-D750A3737BD3")]
    public interface ITimeComponent
    {
        [DispId(1)]
        string GetTime();
    }

    public class TimeComponent : ITimeComponent
    {
        public string GetTime()
        {
            return DateTime.Now.ToString();
        }
    }
}